node/node_test.go

```diff
@@ -244,6 +246,104 @@ func TestRegisterProtocols_OpenRPC_WS(t *testing.T) {
+func TestOpenRPCMetaSchemaSpec(t *testing.T) {
+	stack, err := New(testNodeConfig())
+	if err != nil {
+		t.Fatalf("failed to create protocol stack: %v", err)
+	}
+	defer stack.Close()
+
+	datadir := filepath.Join(os.TempDir(), "node-test")
+	defer os.RemoveAll(datadir)
+
+	stack.config = &DefaultConfig
+	stack.config.HTTPHost = DefaultHTTPHost
+	// enable all available APIs
+	stack.config.HTTPModules = []string{"admin", "debug", "eth", "ethash", "miner", "net", "personal", "rpc", "trace", "txpool", "web3"}
+	stack.config.DataDir = datadir
+
+	if err := stack.Start(); err != nil {
+		t.Fatal(err)
+	}
+	defer stack.Close()
+
+	client, err := rpc.Dial("http://localhost:8545")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer client.Close()
+
+	res := make(map[string]interface{})
+	err = client.Call(&res, "rpc_discover")
+	if err != nil {
+		t.Errorf("client call: %v", err)
+	}
+
+	schemaLoader := gojsonschema.NewSchemaLoader()
+	schemaLoader.Draft = gojsonschema.Draft7 // force schema version
+	schemaLoader.AutoDetect = false
+
+	openRPCSpecFile := "file://./testdata/open-rpc-meta-schema-1.14.0.json"
+	schema, err := schemaLoader.Compile(gojsonschema.NewReferenceLoader(openRPCSpecFile))
+
+	documentLoader := gojsonschema.NewGoLoader(res)
+
+	result, err := schema.Validate(documentLoader)
+	if err != nil {
+		t.Errorf("failed to validate OpenRPC schema: %v", err)
+	}
+
+	if !result.Valid() {
+		for _, desc := range result.Errors() {
+			fmt.Printf("- %s\n", desc)
+		}
+		t.Fatal("the OpenRPC discovery document is not compliant with spec")
+	}
+}
+
+// Suggested latest endpoint to be used is
+// COREGETH_OPENRPC_LATEST_SPEC_ENDPOINT=https://github.com/open-rpc/meta-schema/releases/latest/download/open-rpc-meta-schema.json
+var openRPCSpecLatestTestEnvVarName = "COREGETH_OPENRPC_LATEST_SPEC_ENDPOINT"
+
+// TestOpenRPCMetaSchemaSpecLatest compares local JSON spec file with the latest JSON spec published remotelly,
+// so as we keep the local file in sync
+func TestOpenRPCMetaSchemaSpecLatest(t *testing.T) {
+	openRPCSpecLatestEndpoint := os.Getenv(openRPCSpecLatestTestEnvVarName)
+	if os.Getenv("CI") == "" || openRPCSpecLatestEndpoint == "" {
+		t.Skip()
+	}
+
+	client := &http.Client{}
+	resp, err := client.Get(openRPCSpecLatestEndpoint)
+	if err != nil {
+		t.Fatalf("failed to fetch latest spec JSON: %v", err)
+	}
+	defer resp.Body.Close()
+
+	dec := json.NewDecoder(resp.Body)
+
+	var latestSpec interface{}
+	if err := dec.Decode(&latestSpec); err != nil {
+		t.Fatalf("failed to decode latest spec JSON: %v", err)
+	}
+
+	localJSONFilePath := "./testdata/open-rpc-meta-schema-1.14.0.json"
+	localJSONFile, err := ioutil.ReadFile(localJSONFilePath)
+	if err != nil {
+		t.Fatalf("failed to load local spec JSON file: %v", err)
+	}
+
+	var localSpec interface{}
+	if err := json.Unmarshal([]byte(localJSONFile), &localSpec); err != nil {
+		t.Fatalf("failed to unmarshal local JSON spec: %v", err)
+	}
+
+	if !reflect.DeepEqual(latestSpec, localSpec) {
```

Comment:
Wait, does `reflect.DeepEqual(interface{}{}, interface{}{})` actually provide a good equivalence?

Reply:
Nevermind; experimented with it a little (removing and modifying the local testdata/ file) and running the test; seems ✅.